src/codegen/tests/code_gen_tests.rs

```diff
@@ -493,6 +493,19 @@ fn date_comparisons() {
     insta::assert_snapshot!(result);
 }
 
+#[test]
+#[should_panic]
```

Comment:
Instead of panic, we could use a validation test here. 
What will happen at this point is that the initializer will fail, and then a diagnostic is returned.
You can get this with `codegen_without_unwrap` which will return a single issue.
You can also use another method `codegen_debug_without_unwrap` and pass DebugLevel::None to get a list of all diagnostics produced by the initialization.
You can then verify that there's a diagnostic regarding field `a` which could not be initialized.
I'll try to submit an adjusted test with that soon.